### PR TITLE
RDCC-3118 - PRD6.0 : Ability of solicitors to delete multiple PBAs associated with their organisation (New External API)

### PR DESCRIPTION
**Pull request by viswajit-kuna**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-3118

### Change description ###
New DELETE endpoint to delete the payment accounts of an existing organization

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java [hunk 1]
```diff
@@ -401,6 +402,32 @@ private ProfessionalUser validateInviteUserRequestAndCreateNewUserObject(
...
+
+        //if the organisation is present, check if it is 'ACTIVE'
+        organisationIdentifierValidatorImpl.validateOrganisationIsActive(existingOrganisation);
+
+        //check the pba account number format, remove any blank strings passed
+        //And check if pba belongs to the organisation
+        paymentAccountValidator.validatePaymentAccounts(deletePbaRequest.getPaymentAccounts(), orgId);
+
+        //delete the passed pba account numbers from the organisation
+        paymentAccountService.deletePaymentsOfOrganization(deletePbaRequest, existingOrganisation);
```
Comment (AdilOoze) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java:428]: _Non-important suggestion_: can we update the name of this method `deletePaymentsOfOrganization` to use the English spelling of the word `Organisation` to keep it in line with the other method names?

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/OrganisationService.java [hunk 2]
```diff
@@ -35,5 +35,7 @@ OrganisationResponse updateOrganisation(OrganisationCreationRequest organisation
     void addPbaAccountToOrganisation(Set<String> paymentAccounts, Organisation organisation, boolean pbasValidated,
                                      boolean isEditPba);
 
+    void deletePaymentsOfOrganization(Set<String> paymentAccounts, Organisation organisation);
```
Comment (AdilOoze) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/OrganisationService.java:38]: _Non-important suggestion_: can we update the name of this method `deletePaymentsOfOrganization` to use the English spelling of the word `Organisation` to keep it in line with the other method names?

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/PaymentAccountService.java [hunk 3]
```diff
@@ -9,4 +10,6 @@
     Organisation findPaymentAccountsByEmail(String email);
 
     PbaResponse editPaymentAccountsByOrganisation(Organisation organisation, PbaEditRequest pbaEditRequest);
+
+    void deletePaymentsOfOrganization(DeletePbaRequest deletePbaRequest, Organisation organisation);
```
Comment (AdilOoze) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/PaymentAccountService.java:14]: _Non-important suggestion_: can we update the name of this method `deletePaymentsOfOrganization` to use the English spelling of the word `Organisation` to keep it in line with the other method names?

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/FeatureToggleServiceImpl.java [hunk 4]
```diff
@@ -40,13 +40,14 @@ public void mapServiceToFlag() {
...
         launchDarklyMap.put("OrganisationInternalController.updateOrgMfaStatus", "prd-mfa-flag");
+        launchDarklyMap.put("OrganisationExternalController.deletePaymentAccountsOfOrganisation",
+                "rd-professional-multi-pba");
     }
 
     @Override
     public boolean isFlagEnabled(String serviceName, String flagName) {
         LDUser user = new LDUser.Builder(userName)
             .firstName(userName)
-            .custom("servicename", serviceName)
```
Comment (AdilOoze) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/FeatureToggleServiceImpl.java:49]: is this no longer required?

**Review by AdilOoze — COMMENTED** [review]
Left two non-essential requested changes, please review/address - otherwise happy to approve

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/FeatureToggleServiceImpl.java (hunk 4 above)
Comment (viswajit-kuna) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/FeatureToggleServiceImpl.java:49]: It is not required as `ExUI` is the only consumer of this API. @shreedhar-hmcts Please confirm

**Review by AdilOoze — DISMISSED** [review]

**Review by AdilOoze — DISMISSED** [review]

**Review by shreedhar-hmcts — DISMISSED** [review]

**Review by AdilOoze — DISMISSED** [review]

**Pull request merged**